### PR TITLE
Implement floordiv and NotImplemented.__repr__()

### PR DESCRIPTION
**Pull request by corona10**
1. Implement floordiv.
This PR is related with https://github.com/google/grumpy/issues/269
2. Implement NotImplemented.__repr__()

### CODE REVIEW

File: runtime/int.go [hunk 1]
```diff
@@ -98,6 +98,22 @@ func intEq(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return intCompare(compareOpEq, toIntUnsafe(v), w), nil
 }
 
+func intFloorDiv(f *Frame, v, w *Object) (*Object, *BaseException) {
+	if !v.isInstance(IntType) {
+		return nil, f.RaiseType(TypeErrorType, fmt.Sprintf("descriptor '__floordiv__' requires a 'int' object but received a '%s'", v.typ.Name()))
```
Comment (trotterdylan) [inline runtime/int.go]: The slot delegation code guarantees that the first argument to slot functions is of the correct type, so no need to check here.

File: runtime/int.go [hunk 2]
```diff
@@ -98,6 +98,22 @@ func intEq(f *Frame, v, w *Object) (*Object, *BaseException) {
...
+	}
+
+	if w.isInstance(IntType) {
+		return intDivModOp(f, "__floordiv__", v, w, func(v, w int) (int, divModResult) {
+			return intCheckedDiv(v, w)
+		}, func(z, x, y *big.Int) {
+			longDiv(z, x, y)
+		})
+	}
+	fv, _ := intFloat(f, v)
```
Comment (trotterdylan) [inline runtime/int.go]: You should be able to return NotImplemented in this case and it will delegate to w's operator.

File: runtime/int.go [hunk 3]
```diff
@@ -384,6 +400,7 @@ func initIntType(dict map[string]*Object) {
 	IntType.slots.Div = &binaryOpSlot{intDiv}
 	IntType.slots.DivMod = &binaryOpSlot{intDivMod}
 	IntType.slots.Eq = &binaryOpSlot{intEq}
+	IntType.slots.FloorDiv = &binaryOpSlot{intFloorDiv}
```
Comment (trotterdylan) [inline runtime/int.go]: Thinking about it a little more, I think you can just use intDiv for this slot. I think it does exactly what you want.

File: lib/operator.py [hunk 4]
```diff
@@ -1,33 +1,405 @@
-def eq(a, b):
-  return a == b
+"""
```
Comment (trotterdylan) [inline lib/operator.py]: Seems like this change belongs in a different PR

**Comment by corona10** [comment]
@trotterdylan Thanks for the early review. I will change them as your comment.

**Comment by corona10** [comment]
@trotterdylan Ready for review. PTAL

File: runtime/int.go [hunk 5]
```diff
@@ -471,6 +473,11 @@ func intCheckedDiv(v, w int) (int, divModResult) {
 	return q, r
 }
 
+func intCheckedFloorDiv(v, w int) (int, divModResult) {
```
Comment (trotterdylan) [inline runtime/int.go]: Is this still used somewhere?

Comment (corona10) [inline runtime/int.go]: @trotterdylan 
Not at all. Thanks, I didn't find that! I will update it

**Comment by corona10** [comment]
@trotterdylan PTAL
